servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1258,6 +1267,30 @@ void RenderForwardClustered::_fill_render_list(RenderListType p_render_list, con
 					inst->gi_offset_cache = 0xFFFFFFFF;
 				}
 			}
+
+#ifdef DEBUG_ENABLED
+			if (unlikely(get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_UV2_TEXEL_DENSITY)) {
+				if (inst->data->base_type == RS::INSTANCE_MESH && inst->data->use_baked_light) {
+					Size2 lightmap_size = mesh_storage->mesh_get_lightmap_size_hint(inst->data->base) * inst->data->lightmap_scale;
+
+					if (inst->data->lightmap_size_isp_offset == -1) {
+						int32_t offset = RSG::material_storage->global_shader_uniforms_instance_allocate(inst->data->base);
```

Comment:
Part of the issue here is that that global shader uniforms hashmap takes any RID. I.e. it doesn't check if the RID is for a geometry instance. Here you are passing an RID for the mesh that the Geometry Instance uses. Accordingly, if to MeshInstances use the same mesh, the check fails for the second MeshInstance and the instance uniform does not get updated below. 

I think you will need to store a ``->self`` value in ``RenderGeometryInstanceBase`` so you can access the RID of the GeometryInstance itself

Reply:
Yeah, I was thinking of the same solution.